redis/src/test/scala/zio/redis/SetsSpec.scala

```diff
@@ -758,17 +760,27 @@ trait SetsSpec extends BaseSpec {
           } yield assert(cursor)(isZero) &&
             assert(members)(isNonEmpty)
         },
-        testM("with count over non-empty set with two iterations") {
+        testM("with count over non-empty possibly multiple iterations") {
+
+          // Repeatedly call SCAN until the cursor returns 0. Warning: this brings the entire
+          // set into memory which may not be what you want. It is only done here for test
+          // purposes and we know the Set size.
+          def scanAll(key: String, cursor: Long, regex: Option[Regex], count: Option[Count], members: Set[String])
```

Comment:
Think about using `repeatWhile/repeatUntil` method.

Reply:
9ne thing that comes to mind is to use ZStream.paginateM with one collect in the end to accumulate results.

Reply:
For example:

```scala
def scanAll(key: String) = {
  ZStream
    .paginateChunkM(0L) { cursor =>
      sScan(key, cursor).map {
        case (nc, _) if nc == 0 => (Chunk.empty, None)
        case (nc, nm)           => (nm, Some(nc))
      }
    }
    .runCollect
}

Reply:
Thanks, I'll try this. Is it preferable to bring in ZStreams or use just ZIO here? Personally, I think because Redis and streaming go so well together it is fine to have that dependency in the tests

Reply:
Ah, we already have it anyhow, executor is built on top of streams :).